### PR TITLE
enhance: streaming node client implementation

### PR DESCRIPTION
**Pull request by chyezh**
issue: #33285

- add streaming node grpc client wrapper
- add unittest for streaming node grpc client side
- fix binary unsafe bug for message

### CODE REVIEW

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by jaime0815** [comment]
todo: refine connection granularity of consumer stream to pchannel  level

File: internal/streamingnode/client/handler/handler_client_impl.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	})
+	if err != nil {
+		return nil, err
+	}
+	return c.(Consumer), nil
+}
+
+// createHandlerUntilStreamingNodeReady creates a handler until streaming node ready.
+// If streaming node is not ready, it will block until new assignment term is coming or context timeout.
+func (hc *handlerClientImpl) createHandlerUntilStreamingNodeReady(ctx context.Context, pchannel string, create func(ctx context.Context, assign *types.PChannelInfoAssigned) (any, error)) (any, error) {
```
[the remark this answers is not in the document]
Comment (chyezh) [reply internal/streamingnode/client/handler/handler_client_impl.go:89]: will fix it soon

**Comment by chyezh** [comment]
rerun ut

File: internal/streamingnode/client/handler/assignment/watcher_impl.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	defer w.cond.L.Unlock()
+
+	if info, ok := w.assignments[channel]; ok {
+		return &info
+	}
+	return nil
+}
+
+// Watch watches the channel assignment.
+func (w *watcherImpl) Watch(ctx context.Context, channel string, previous *types.PChannelInfoAssigned) error {
```
[the remark this answers is not in the document]
Comment (chyezh) [reply internal/streamingnode/client/handler/assignment/watcher_impl.go:83]: got it

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Pull request merged**